Frameworks/CoreGraphics/CGContext.mm

```diff
@@ -1242,6 +1264,34 @@ void CGContextClearRect(CGContextRef context, CGRect rect) {
     renderTarget->PopAxisAlignedClip();
 }
 
+static bool __CGContextCreateShadowEffect(CGContextRef context, ID2D1DeviceContext* deviceContext, ID2D1Image* inputImage, ID2D1Effect** outShadowEffect) {
```

Comment:
@aballway It exposes an interface more like classic COM APIs, which don't consume C++ types.